File: first-non-duplicate.rb
# Write a function that returns the first non-duplicated character in a string

def first_non_duplicate(string)
  letter_hash = {}
  string.each_char do |char|
    if letter_hash[char]
      letter_hash[char] += 1
    else
      letter_hash[char] = 1
    end
  end
  letter_hash.each do |key, value|
    if value == 1
      return key
    end
  end
end

p first_non_duplicate("minimum") == "n" # true

File: first-non-duplicate.js
// Write a function that returns the first non-duplicated character in a string

function firstNonDuplicate(string) {
  let letterHash = {};
  for (let i = 0; i < string.length; i++) {
    if (letterHash[string[i]]) {
      letterHash[string[i]]++;
    } else {
      letterHash[string[i]] = 1;
    }
  }
  for (let j = 0; j < string.length; j++) {
    if (letterHash[string[j]] == 1) {
      return string[j];
    }
  }
}

console.log(firstNonDuplicate("minimum"));
